functions/openhab.js

```diff
@@ -151,6 +168,59 @@ function getItemsState(request,response) {
 		})
 }
 
+/**
+ * Gets Temperature or Thermostat Data
+ */
+function getTempData(item) {
+	var thermData = {};
+	var thermItems = item.tags.toString().includes("Thermostat") ? getThermostatItems(item.members) : getThermostatItems([item]);
+	
+	//Are we dealing with Fahrenheit?
+	var isF = item.tags.indexOf('Fahrenheit') >= 0 ? true : false;
+	
+	//store long json variables in easier variables to work with below
+	var tstatMode = thermItems.hasOwnProperty('heatingCoolingMode') ? (thermItems.heatingCoolingMode.state.length == 1 ? utils.normalizeThermostatMode(thermItems.heatingCoolingMode.state) : thermItems.heatingCoolingMode.state) : 'heat'
+	var currTemp =  thermItems.hasOwnProperty('currentTemperature') ? (isF ? utils.toC(thermItems.currentTemperature.state) : thermItems.currentTemperature.state) : '';
+	var tarTemp = thermItems.hasOwnProperty('targetTemperature') ? (isF ? utils.toC(thermItems.targetTemperature.state) : thermItems.targetTemperature.state) : '';
+	var curHum = thermItems.hasOwnProperty('currentHumidity') ? thermItems.currentHumidity.state : '';
+
+	//We will always report device as "online", not sure what this does, but per docs it seems necessary
+	thermData.online = true;
+
+	//populate only the necessary json values, otherwise GA will get confused if keys are empty
+	if (item.tags.toString().includes("Thermostat")) {
+		thermData.thermostatMode = tstatMode;
+		if (thermItems.hasOwnProperty('currentTemperature')) thermData.thermostatTemperatureAmbient = Number(parseFloat(currTemp).toFixed(1));
+		if (thermItems.hasOwnProperty('targetTemperature')) thermData.thermostatTemperatureSetpoint = Number(parseFloat(tarTemp).toFixed(1));
+		if (thermItems.hasOwnProperty('currentHumidity')) thermData.thermostatHumidityAmbient = Number(parseFloat(curHum).toFixed(0));              
+	}
+	else if (item.tags.toString().includes("CurrentTemperature")) {
+		thermData.thermostatMode = "heat"; // doesn't matter, GA will only state the number (since we force Ambient and Setpoint to match)
+		if (thermItems.hasOwnProperty('currentTemperature')) thermData.thermostatTemperatureAmbient = Number(parseFloat(currTemp).toFixed(1));
+		thermData.thermostatTemperatureSetpoint = thermData.thermostatTemperatureAmbient;
+	}
+	
+	return thermData;
+}
+
+function getLightData(item) {
+  console.log(item.name + " State: " + item.state);
+  return {
+	on: item.state === 'ON' ? true : (item.state > 0 ? true : false),
```

Comment:
`item.state > 0 ? true : false` can be replaced with simply `item.state` or `!!item.state` if you need to return boolean value.

Reply:
I am not sure I follow.  Google requires a boolean response to the "on" key.  I appended the code because it would only work with switches that return ON and OFF (which it's pretty simple to do with ternary statement).  However, Dimmer items return a number 0-100.  So I was trying to do both here, which works.  If there's a simpler more efficient way if you can edit I'd appreciate.